app/src/main/java/edu/mit/mitmobile2/shuttles/fragment/ShuttleStopFragment.java

```diff
@@ -182,6 +182,19 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
         int realPosition = stopViewPagerAdapter.getRealPosition(currentPosition);
         stopViewPagerAdapter.updatePredictions(realPosition, stops.get(realPosition).getPredictions());
 
+        //Update next and previous stops' predictions as well due to view pager storing adjacent fragments
+        int previousPosition = realPosition - 1;
+        if (previousPosition == -1) {
```

Comment:
It'd make sense to use the modulo operator here and below to handle the wrapping of stops

So something like int previousPosition = --realPosition % stops.size();